components/ds3231/ds3231.c

```diff
@@ -87,6 +87,30 @@ static uint8_t dec2bcd(uint8_t val)
     return ((val / 10) << 4) + (val % 10);
 }
 
+#include <stdio.h>
+
+// Function to check if a year is a leap year
+int isLeapYear(int year) { return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0); }
+
+// Function to convert year, month, and day to days since January 1st
+int daysSinceJanuary1st(int year, int month, int day)
+{
+    int daysPerMonth[] = { 0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
```

Comment:
1. CamelCase
2. This is constant, so declare it `static const ....`

Reply:
The code is updated to use them as static const arrays.